app/lib/models/animal_transport_record.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:app/models/shipper_info.dart';
+import 'package:app/models/transporter_info.dart';
+import 'package:flutter/material.dart';
+
+import 'acknowledgement_info.dart';
+import 'contingency_plan_info.dart';
+import 'delivery_info.dart';
+import 'feed_water_rest_info.dart';
+import 'loading_vehicle_info.dart';
+
+@immutable
+class AnimalTransportRecord {
+  final ShipperInfo shipInfo;
+  final TransporterInfo tranInfo;
+  final FeedWaterRestInfo fwrInfo;
+  final LoadingVehicleInfo vehicleInfo;
+  final DeliveryInfo deliveryInfo;
+  final AcknowledgementInfo ackInfo;
+  final ContingencyPlanInfo contingencyInfo;
+
+  AnimalTransportRecord(this.shipInfo, this.tranInfo, this.fwrInfo,
+      this.vehicleInfo, this.deliveryInfo, this.ackInfo, this.contingencyInfo);
+
+  AnimalTransportRecord withShipInfo(ShipperInfo newShipInfo) =>
```

Comment:
This is a creator method (uses `this` fields) and not a factory, which is like a static creator.

Reply:
This is good to have, as these methods will support splitting the ATR editing across screens, and also still support editing in once screen with accordion menus.